src/Microsoft.OpenApi.OData.Reader/Operation/RefDeleteOperationHandler.cs

```diff
@@ -40,8 +40,22 @@ protected override void SetBasicInfo(OpenApiOperation operation)
             // OperationId
             if (Context.Settings.EnableOperationId)
             {
-                string prefix = "DeleteRef";
-                operation.OperationId = GetOperationId(prefix);
+                string prefix = "DeleteRef";
+                var segments = GetOperationId().Split('.').ToList();
```

Comment:
```suggestion
                var segments = GetOperationId().Split('.').ToList();
                if (segments.Length <= 0) return;
```

Reply:
In this case... `if (segments.Count <= 0) return;`

Reply:
I think part of your comment is missing?

Reply:
I believe the `if (segments.Length <= 0) return;` is technically unreachable code. As `Split` will still return the original string as the only item in the collection if it does not contain the split character.

Reply:
right, it only works if remove empty entries is passed
https://dotnetfiddle.net/JGpHaR
Resolving